### ISSUE TITLE
Authentication fails using ibm-credentials.env file

### ISSUE THREAD

**Issue by esb** [watson-developer-cloud/ruby-sdk#73]
**Describe the bug**
Authentication always fails when trying to use the downloaded ibm-credentials.env file.

**To Reproduce**
Create a IBM_CREDENTIALS_FILE environment variable and point to the downloaded credentials file. Simple create a service using only the version parameter, and it will fail to authenticate.

**Expected behavior**
No authentication errors.

**Problem Resolution**
The bug is pretty simple to fix. The ibm_cloud_sdk_core uses the display_name parameter to load the authentication variables from the file. If this parameter is not set, then the credentials file is not loaded.

All of the initialization routines in this package set the display_name parameter AFTER the call to super, so the value is never passed to the super class (ibm_cloud_sdk_core). The display_name parameter needs to be set before calling super.

As a side comment, even though this fixes the loading and parsing of the credentials file, code is missing from the base_service module to actually use the apikey and url read from the credentials file. It looks like this functionality was never actually finished.

**Comment by mamoonraja** [comment]
Thanks a lot for opening the issue, I am debugging and will have a fix out soon.

**Comment by mamoonraja** [comment]
The issue has been fixed in `0.19.1`, I have tested the functionality by downloading the credentials file and trying it with the following snippet:

```
assistant = IBMWatson::AssistantV1.new(
  version: "2018-02-16"
)
response = assistant.list_workspaces.result
puts JSON.pretty_generate(response)
```

Feel free to open more issues or even contribute with a PR :) Please let us know if you have any more feedback as we are targeting our first major release in https://github.com/watson-developer-cloud/ruby-sdk/issues/56.
